### ISSUE TITLE
CI warnings: Use Custom Errors instead of require statements

### ISSUE THREAD

**Issue by dcroote** [api3dao/airnode#1870]
Our CI is generating warnings like the following on the [latest master run](https://github.com/api3dao/airnode/actions/runs/5985018642):

```
Build and lint Airnode: packages/airnode-examples/contracts/authenticated-coinmarketcap/Requester.sol#L36
Use Custom Errors instead of require statements
```
The corresponding code:
https://github.com/api3dao/airnode/blob/c59f68e2c5439921d7d9081b0345b171a29f6eaf/packages/airnode-examples/contracts/authenticated-coinmarketcap/Requester.sol#L36

There are 10 in total, all corresponding to similar `Requester.sol` contracts within `airnode-examples`. 

Do we want to make the suggested change?

**Comment by bbenligiray** [comment]
Nah, that's just the sheeple bleating, custom errors vs. string errors is a tradeoff. String errors work on all chains and block explorers (people don't care because they're not on as many weird chains as us), even when the client doesn't know the contract ABI. This comes at the cost of higher deployment cost and higher unhappy path gas cost, so I only use custom errors if the contract is expected to be deployed many times and revert very frequently (as in https://github.com/api3dao/merkle-funder/blob/main/contracts/MerkleFunder.sol).

Immutables being capitalized SNAKE_CASE is flat out wrong (https://docs.soliditylang.org/en/v0.8.21/contracts.html#constant-and-immutable-state-variables) so I don't know what the solhint guys were thinking there.

I'm opening a PR to ignore both of these warnings
